mne_bids/tests/test_write.py

```diff
@@ -883,13 +883,37 @@ def test_eegieeg(dir_name, fname, reader, _bids_validate):
     data_path = op.join(testing.data_path(), dir_name)
     raw_fname = op.join(data_path, fname)
 
+    # the BIDS path for test datasets to get written to
+    bids_path = _bids_path.copy().update(root=bids_root, datatype='eeg')
+
+    # read in the source raw file
     raw = reader(raw_fname)
 
+    # if there were events in the original raw dataset that
+    # were not encoded in the stim channel,
+    # they should be written even if write_raw_bids does not
+    # pass in events
+    events, events_id = mne.events_from_annotations(raw, event_id=None)
+    if events_id != {} and 'stim' not in raw:
```

Comment:
I don't get why this `if events_id != {} and 'stim' not in raw:`

to me it should be simpler. If you have annotations present with raw it's written and you read
the written events as an annotation to make sure it's correct.

bids events.tsv is very close to an mne annotations written to disk as .txt or csv

Reply:
Ah okay I got rid of this `if` statement. I don't think it's necessary anymore and I took it out and the tests worked.